bindgen/src/templates/common/Types.kt

```diff
@@ -16,8 +16,22 @@ interface Disposable : AutoCloseable {
     companion object {
         internal fun destroy(vararg args: Any?) {
             for (arg in args) {
-                if (arg is Disposable) {
-                    arg.destroy()
+                when (arg) {
+                    is Disposable -> arg.destroy()
+                    is List<*> -> {
```

Comment:
How do you feel about doing something higher up like a [Iterable](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.collections/-iterable/) or [Collection](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.collections/-collection/) so we catch things like Sets. We might also need Array because it doesn't have a parent class.

Reply:
I added branches handling such types as well. Though handling Iterable, Collection, or Array won't be used yet since the bindgen uses `List<*>` or `Map<*, *> for collections only. Ideally, `Disposable.destroy` should be replaced with static type checks in the bindgen and call `.destroy()` or `?.destroy()` only when needed. `Disposable.destroy()` is there only because checks for compound types make the template way more complicated.